src/NuGet.Core/NuGet.Commands/RestoreCommand/SourceRepositoryDependencyProvider.cs

```diff
@@ -682,6 +682,13 @@ private async Task LogErrorAsync(ILogger logger, string id, FatalProtocolExcepti
                 else
                 {
                     await logger.LogAsync(RestoreLogMessage.CreateError(NuGetLogCode.NU1301, e.Message, id));
+                    Exception exception = e as Exception;
+
+                    while (exception.InnerException != null)
+                    {
+                        await logger.LogAsync(level: LogLevel.Verbose, e.InnerException.Message);
```

Comment:
Ideally you want these in a single log message. 

The rule of thumb is that each log message should have an error code. This causes additional messages to be included. If this happens for multiple projects, the fact which messages refer to which issues can be lost.

Reply:
We really don't know what exactly the message from the exception is; would Nu1301 be appropriate for this?

Reply:
just used `ExceptionUtilities.DisplayMessage(e)` instead. It makes it a single log message.